packages/next/build/entries.ts

```diff
@@ -136,7 +141,11 @@ export function createEntrypoints(
     const serverBundlePath = posix.join('pages', bundleFile)
 
     const isLikeServerless = isTargetLikeServerless(target)
-    const isFlight = isFlightPage(config, absolutePagePath)
+    const isReserved = isReservedPage(page)
+    const isCustomError = isCustomErrorPage(page)
+    const isFlight =
+      hasServerComponents && isFlightPage(config, absolutePagePath)
```

Comment:
I think it already checks the `config.experimental.concurrentFeatures && config.experimental.serverComponents` logic inside `isFlightPage`.